### PR TITLE
CF-USB acceptance-tests-brain updates

### PR DESCRIPTION
**Pull request by mook-as**
## Description

This updates our acceptance-tests-brain tests for CF-USB:
- Refactor test to use helper class
- Add tests for postgres
- Add ability to override versions to use
- Actually test using the helm charts, rather than the raw docker images; the customers are expected to use the charts.

## Test plan

Locally tested with overrides:

- MySQL:
> `env.VERBOSE=1 env.MYSQL_REPO=https://mook-as.github.io/kubernetes-charts-suse-com/ env.MYSQL_CHART_VERSION=1.0.1-9-geeea5b9 env.HELM_REGISTRY_HOSTNAME=index.docker.io`

- Postgres:
> `env.VERBOSE=1 env.POSTGRES_REPO=https://mook-as.github.io/kubernetes-charts-suse-com/ env.POSTGRES_CHART_VERSION=1.0.1-9-geeea5b9 env.HELM_REGISTRY_HOSTNAME=index.docker.io`

The temporary helm chart repository is at [mook-as/kubernetes-charts-suse-com](https://github.com/mook-as/kubernetes-charts-suse-com) (`gh-pages` branch; the `master` branch is the expanded charts).  This is not part of the code we will be pushing.

This should also succeed on Jenkins.

### CODE REVIEW

**Comment by mook-as** [comment]
Force-pushed to add the missing postgres bits.

**Comment by andreas-kupries** [comment]
>This should also succeed on Jenkins.

Jenkins apparently did not retrigger on the force pushed commit adding the missing postgres bits.
Manually triggered now.

**Comment by andreas-kupries** [comment]
Build still fails. Still complains about missing postgres things in the `compile` stage.

**Comment by mook-as** [comment]
Ah, right, Jenkins uses SLE for builds.  That doesn't have all the repos (actually, it looks like it's just `obs://Cloud:Platform:SUSE-Stemcell`), so installing postgres doesn't work (not that we should be doing it this way to begin with…)

I'll need to think of something else.

File: src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+##    create a service from it, and check that this service is viewable too.
+#
+## In the code below these phases are marked with "--(N)--" where N is
+## the step number.
+#
+## Note, the applications of step 1 and 2 are docker apps. This is why
+## the `pre-start.sh` script enables the `diego_docker` feature-flag
+## of CF. For step (3) the `pre-start.sh` script extended the `cf`
+## client with the `cf-usb-plugin` plugin.
+
```
Comment (andreas-kupries) [inline src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb]: Should possibly document what methods derived classes are expected to define, i.e. the hooks the helper invokes for the engine specific actions.

Comment (mook-as) [reply src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb]: Fixed in 8c5d663aaab02b17417acd57003a6ffaf66d00ac

File: src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+require_relative 'testutils'
+require 'json'
+
+class CFUSBTestBase
+    MYSQL_USER = 'root'
+    MYSQL_PASS = 'testpass'
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb]: Fixed in 8c5d663aaab02b17417acd57003a6ffaf66d00ac

File: src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    def deploy_server
+        raise NotImplementedError, 'The derived class is expected to implement this'
+    end
+
+    ## --(2)-- Create and configure the sidecar for usb.
+    def deploy_sidecar
+        raise NotImplementedError, 'The derived class is expected to implement this'
+    end
+
+    # --(3)-- Create a driver endpoint to the mysql sidecar (== service type)
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/scf-release/src/acceptance-tests-brain/test-scripts/cf_usb_helper.rb]: Fixed in 8c5d663aaab02b17417acd57003a6ffaf66d00ac

**Pull request merged**